pkg/controllers/nodetemplate/hash_test.go

```diff
@@ -0,0 +1,64 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodetemplate_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/aws-sdk-go/aws"
+
+	. "github.com/aws/karpenter-core/pkg/test/expectations"
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+)
+
+var _ = Describe("AWSNodeTemplate Static Drift Hash", func() {
+	It("should update the static drift hash when nodeTemplate static field is updated", func() {
+		ExpectApplied(ctx, env.Client, nodeTemplate)
+		ExpectReconcileSucceeded(ctx, controller, client.ObjectKeyFromObject(nodeTemplate))
+		nodeTemplate = ExpectExists(ctx, env.Client, nodeTemplate)
+
+		expectedHash := nodeTemplate.Hash()
+		Expect(nodeTemplate.ObjectMeta.Annotations[v1alpha1.AnnotationNodeTemplateHash]).To(Equal(expectedHash))
+
+		nodeTemplate.Spec.UserData = aws.String("test-userData-2")
+		ExpectApplied(ctx, env.Client, nodeTemplate)
+		ExpectReconcileSucceeded(ctx, controller, client.ObjectKeyFromObject(nodeTemplate))
+		nodeTemplate = ExpectExists(ctx, env.Client, nodeTemplate)
+
+		expectedHashTwo := nodeTemplate.Hash()
+		Expect(expectedHash).ToNot(Equal(expectedHashTwo))
+		Expect(nodeTemplate.ObjectMeta.Annotations[v1alpha1.AnnotationNodeTemplateHash]).To(Equal(expectedHashTwo))
+	})
+	It("should not update the static drift hash when nodeTemplate behavior/dynamic field is updated", func() {
+		ExpectApplied(ctx, env.Client, nodeTemplate)
+		ExpectReconcileSucceeded(ctx, controller, client.ObjectKeyFromObject(nodeTemplate))
+		nodeTemplate = ExpectExists(ctx, env.Client, nodeTemplate)
+
+		expectedHash := nodeTemplate.Hash()
+		Expect(nodeTemplate.ObjectMeta.Annotations[v1alpha1.AnnotationNodeTemplateHash]).To(Equal(expectedHash))
+
+		nodeTemplate.Spec.SubnetSelector = map[string]string{"aws-ids": "subnet-test1"}
+		nodeTemplate.Spec.SecurityGroupSelector = map[string]string{"aws-ids": "sg-test1"}
+		nodeTemplate.Spec.AMISelector = map[string]string{"ami-test-key": "ami-test-value"}
```

Comment:
Make sure you add in behavioral fields in this test too.

Reply:
These are all the fields that will not be drifted